Acceptance test destroy steps are running MigrateStates

When we update the provider SDK to 0.12-alpha4, our test destroy steps are running through a MigrateState. This only seems to happen in tests: destroying resources with a locally-built binary works fine.

To reproduce, start with a branch that has the 0.12 SDK vendored in, like https://github.com/terraform-providers/terraform-provider-google/tree/gomod-update-sdk. Then run a test that creates a compute instance:
```
$ make testacc TEST=./google TESTARGS='-run=TestAccComputeAddress_instanceWithIpExample'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google -v -run=TestAccComputeAddress_instanceWithIpExample -timeout 120m -ldflags="-X=github.com/terraform-providers/terraform-provider-google/version.ProviderVersion=acc"
=== RUN   TestAccComputeAddress_instanceWithIpExample
=== PAUSE TestAccComputeAddress_instanceWithIpExample
=== CONT  TestAccComputeAddress_instanceWithIpExample
--- FAIL: TestAccComputeAddress_instanceWithIpExample (41.32s)
    testing.go:605: Error destroying resource! WARNING: Dangling resources
        may exist. The full state and error is shown below.

        Error: Error refreshing: migration error: found disk.# value in unexpected format: strconv.Atoi: parsing "": invalid syntax

        State: <nil>
FAIL
FAIL	github.com/terraform-providers/terraform-provider-google/google	41.675s
make: *** [testacc] Error 1
```

This error message comes from https://github.com/terraform-providers/terraform-provider-google/blob/gomod-update-sdk/google/resource_compute_instance_migrate.go, and when running with debug logs on we can see that the migrate function was called starting with a state version of 0, even though during the test itself, it would get written with a state version of 6 (the most recent version). This causes the destroy step to fail since it's expecting values in state that don't get written with higher schema versions.

--- a/helper/resource/state_shim.go
+++ b/helper/resource/state_shim.go
@@ -88,6 +88,11 @@ func shimNewState(newState *states.State, schemas *terraform.Schemas) (*terrafor
 						Tainted:    i.Current.Status == states.ObjectTainted,
 					},
 				}
+				if i.Current.SchemaVersion != 0 {
+					resState.Primary.Meta = map[string]interface{}{
+						"schema_version": i.Current.SchemaVersion,
+					}
+				}
 
 				for _, dep := range i.Current.Dependencies {
 					resState.Dependencies = append(resState.Dependencies, dep.String())
@@ -118,6 +123,11 @@ func shimNewState(newState *states.State, schemas *terraform.Schemas) (*terrafor
 						ID:         flatmap["id"],
 						Attributes: flatmap,
 						Tainted:    dep.Status == states.ObjectTainted,
+					}
+					if dep.SchemaVersion != 0 {
+						deposed.Meta = map[string]interface{}{
+							"schema_version": dep.SchemaVersion,
+						}
 					}
 
 					resState.Deposed = append(resState.Deposed, deposed)

--- a/helper/resource/state_shim_test.go
+++ b/helper/resource/state_shim_test.go
@@ -3,9 +3,8 @@ package resource
 import (
 	"testing"
 
-	"github.com/hashicorp/terraform/configs/configschema"
-
 	"github.com/hashicorp/terraform/addrs"
+	"github.com/hashicorp/terraform/configs/configschema"
 	"github.com/hashicorp/terraform/states"
 	"github.com/hashicorp/terraform/terraform"
 	"github.com/zclconf/go-cty/cty"
@@ -29,8 +28,9 @@ func TestStateShim(t *testing.T) {
 			Name: "foo",
 		}.Instance(addrs.NoKey),
 		&states.ResourceInstanceObjectSrc{
-			Status:    states.ObjectReady,
-			AttrsFlat: map[string]string{"id": "foo", "bazzle": "dazzle"},
+			Status:        states.ObjectReady,
+			AttrsFlat:     map[string]string{"id": "foo", "bazzle": "dazzle"},
+			SchemaVersion: 7,
 			Dependencies: []addrs.Referenceable{
 				addrs.ResourceInstance{
 					Resource: addrs.Resource{
@@ -208,6 +208,9 @@ func TestStateShim(t *testing.T) {
 							Attributes: map[string]string{
 								"id":     "foo",
 								"bazzle": "dazzle",
+							},
+							Meta: map[string]interface{}{
+								"schema_version": 7,
 							},
 						},
 						Dependencies: []string{"test_thing.baz"},

